### PR TITLE
Print mnesia:wait_for_tables progress

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "MIM-2351 Better logging when something takes too long".

Proposed changes include:
* Print log message with waiting tables to load, print table info for them.
* Repeat waiting.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/248526) / elasticsearch_and_cassandra_mnesia / 199950e29c8faa990b9bcf04796df3ab6e7234df
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4427/248526/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4427/248526/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40f128849e8e09.2024-12-09_15.22.09/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/249107) / elasticsearch_and_cassandra_mnesia / c6131e5a18e841e3a44096705455c83b1d2b0ace
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4427/249107/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4427/249107/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40cb73444d6dc2.2024-12-11_12.36.39/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: test/common/mongoose_helper.erl [hunk 1]
```diff
@@ -351,18 +351,20 @@ inject_module(Module) ->
 inject_module(Module, ReloadIfAlreadyLoaded) ->
     inject_module(mim(), Module, ReloadIfAlreadyLoaded).
 
--spec inject_module(Node :: atom(),
+-spec inject_module(Node :: distributed_helper:rpc_spec(),
                     Module :: module(),
                     ReloadIfAlreadyLoaded :: no_reload | reload) ->
     ok | already_loaded.
-inject_module(Node, Module, no_reload) ->
+inject_module(#{node := NodeAtom} = Node, Module, _) when NodeAtom =:= node() ->
```
Comment (jacekwegr) [inline test/common/mongoose_helper.erl:358]:
```suggestion
inject_module(#{node := NodeAtom} = _Node, _, _) when NodeAtom =:= node() ->
```

**Review by jacekwegr — COMMENTED** [review]
Thanks you for adding the tests! I just have a few minor comments.

**Review by jacekwegr — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**